lib/uuid_mixin.rb

```diff
@@ -1,17 +1,17 @@
 module UuidMixin
   extend ActiveSupport::Concern
   included do
-    before_validation :set_guid, :on => :create if respond_to?(:before_validation)
+    default_value_for(:guid, :allows_nil => false) { SecureRandom.uuid }
+    validates :guid,  :uniqueness => true, :presence => true
   end
 
-  private
-
-  def set_guid
-    self.guid ||= SecureRandom.uuid if self.respond_to?(:guid) && self.respond_to?(:guid=)
```

Comment:
Do we have to worry about removing this guard for including this in classes that don't actually have a `guid` attribute?

Reply:
Why would this mixin be included in anything that doesn't have a `guid` attribute?

Reply:
I would hope it wouldn't be. I'm just pointing out that this check has been removed in this patch.